Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/shape_smoothing.h

```diff
@@ -0,0 +1,408 @@
+{
+  using boost::choose_param;
+  using boost::get_param;
+
+  #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+  CGAL::Timer t;
+  std::cout << "Smoothing parameters...";
+  std::cout.flush();
+  t.start();
+  #endif
+
+  typedef typename GetGeomTraits<PolygonMesh, NamedParameters>::type GeomTraits;
+
+  typedef typename GetVertexPointMap<PolygonMesh, NamedParameters>::type VertexPointMap;
+  VertexPointMap vpmap = choose_param(get_param(np, internal_np::vertex_point),
+                               get_property_map(CGAL::vertex_point, pmesh));
+
+  typedef typename boost::graph_traits<PolygonMesh>::vertex_descriptor vertex_descriptor;
+  typedef typename boost::lookup_named_param_def <
+      internal_np::vertex_is_constrained_t,
+      NamedParameters,
+      internal::Constrained_vertices_map<vertex_descriptor>
+    > ::type VCMap;
+  VCMap vcmap = choose_param(get_param(np, internal_np::vertex_is_constrained),
+                             internal::Constrained_vertices_map<vertex_descriptor>());
+
+  std::size_t nb_iterations = choose_param(get_param(np, internal_np::number_of_iterations), 1);
+
+  #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+  t.stop();
+  std::cout << "\rSmoothing parameters done ("<< t.time() <<" sec)" << std::endl;
+  std::cout << "Remesher construction...";
+  std::cout.flush();
+  t.reset(); t.start();
+  #endif
+
+  internal::Curvature_flow<PolygonMesh, VertexPointMap, VCMap, GeomTraits>
+          curvature_remesher(pmesh, vpmap, vcmap);
+
+  #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+  t.stop();
+  std::cout << " done ("<< t.time() <<" sec)." << std::endl;
+  std::cout << "Removing degenerate faces..." << std::endl;
+  t.reset(); t.start();
+  #endif
+
+  curvature_remesher.remove_degenerate_faces();
+
+  #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+  t.stop();
+  std::cout << " done ("<< t.time() <<" sec)." << std::endl;
+  std::cout << "Initializing..." << std::endl;
+  t.reset(); t.start();
+  #endif
+
+  curvature_remesher.init_smoothing(faces);
+
+  #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+  t.stop();
+  std::cout << " done ("<< t.time() <<" sec)." << std::endl;
+  std::cout << "#iter = " << nb_iterations << std::endl;
+  std::cout << "Shape smoothing..." << std::endl;
+  t.reset(); t.start();
+  #endif
+
+  for(std::size_t i=0; i<nb_iterations; ++i)
+  {
+
+    #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+        std::cout << " * Iteration " << (i + 1) << " *" << std::endl;
+    #endif
+
+    curvature_remesher.curvature_smoothing();
+
+  }
+
+  #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+  t.stop();
+  std::cout << "Shape smoothing done in ";
+  std::cout << t.time() << " sec." << std::endl;
+  std::cout<<std::endl;
+  #endif
+}
+
+/*!
+* \ingroup PMP_meshing_grp
+* smooths the overall shape of the mesh by using the mean curvature flow.
+* The effect depends on the curvature of each area and on a time step which
+* represents the amount by which vertices are allowed to move.
+* The result is a conformal map of the initial surface and converges to a sphere.
```

Comment:
"is a conformal map" should be "conformally maps"
Imho, the result is a mesh

Reply:
ping @kkatrio